gradle/javadoc.gradle

```diff
@@ -18,17 +18,18 @@ import static org.gradle.util.CollectionUtils.single
 
 
 def javaMainProjects = subprojects.findAll { !['util'].contains(it.name) }
-task docs(type: Javadoc) {
-    destinationDir = new File(rootDir, 'build/docs')
-    javaMainProjects.each { proj ->
-        proj.tasks.withType(Javadoc).each { javadocTask ->
-            source += javadocTask.source
-            classpath += javadocTask.classpath
-            excludes += javadocTask.excludes
-            includes += javadocTask.includes
-            dependsOn +=javadocTask.dependsOn
-        }
-    }
+
+task docs {
+    dependsOn subprojects.collect { it.tasks.withType(Javadoc) }
+    dependsOn subprojects.collect { it.tasks.withType(ScalaDoc) }
+}
+
+def buildDocsUrl(String version, String packageName) {
+    def docVersion = version.split('-').head()
+    def versionList = docVersion.split('\\.')
+    def patch = (versionList[1] as int) - 1
+    docVersion = patch >= 0 ? "${versionList[0]}.${patch}" : '4.0'
```

Comment:
I'm not following the logic.   What's the intention here?

Reply:
This is to reduce the minor version by 1 and then set the url eg: 4.1.0-SNAPSHOT will link to 4.0 docs.  The idea is these docs will actually exist. So if some loads a SNAPSHOT then the javadoc.jars will not contain invalid links.